### PR TITLE
Add back Diagonal DPI

### PR DESCRIPTION
**Pull request by 1bsyl**
After https://github.com/libsdl-org/SDL/pull/7158

### CODE REVIEW

File: src/sdl2_compat.c [hunk 1]
```diff
@@ -4183,6 +4183,35 @@ DECLSPEC int SDLCALL SDL_ConvertAudio(SDL_AudioCVT *cvt)
...
+    if (vdpi) {
+        *vdpi = v;
+    }
+    if (ddpi) {
+        SDL2_DisplayMode mode;
+        float diagonal_dot, diagonal_inch;
+        if (SDL_GetCurrentDisplayMode(displayIndex, &mode) < 0) {
+            return -1;
+        }
+        diagonal_inch = SDL_sqrt(((float)mode.w / h) * ((float)mode.w / h) + ((float)mode.h / v) * ((float)mode.h / v));
```
Comment (sezero) [inline src/sdl2_compat.c:4206]: SDL3_sqrt

File: src/sdl2_compat.c [hunk 2]
```diff
@@ -4183,6 +4183,35 @@ DECLSPEC int SDLCALL SDL_ConvertAudio(SDL_AudioCVT *cvt)
...
+        diagonal_dot = SDL_sqrt(mode.w * mode.w + mode.h * mode.h);
```
Comment (sezero) [inline src/sdl2_compat.c:4207]: SDL3_sqrt

**Comment by 1bsyl** [comment]
I wonder also, if I should use SDL_GetCurrentDisplayMode or SDL_GetDisplayBounds ...

**Comment by 1bsyl** [comment]
updated to use SDL3_sqrtf

**Pull request closed without merging**